lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -240,6 +265,11 @@ async sub recursive_search {
 
     my $current_block = Math::BigInt->from_hex(await $self->rpc_client->get_last_block());
 
+    unless ($current_block->bgt(0)) {
+        my $syncing = await $self->rpc_client->syncing();
+        $current_block = Math::BigInt->from_hex($syncing->{currentBlock}) if $syncing;
```

Comment:
This logic seems generally suspect.  It feels like `get_last_block` should handle any validation and potentially return the `syncing`-derived value itself.  It should either return the "true last block" when possible or, at least, have a different name which indicates what it is actually returning.

It's hard for me to imagine a situation where this `syncing`-check logic isn't required  or helpful after checking `get_last_block`.

Reply:
`eth_blockNumber` will return empty when the node is syncing, but I think is better to remove this from here, we don't want this running with a non synced node, I will remove the syncing routine.